core/src/providers/chat_messages.rs

```diff
@@ -82,6 +96,8 @@ pub struct AssistantChatMessage {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub name: Option<String>,
     pub role: ChatMessageRole,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub contents: Option<Vec<AssistantContentItem>>,
```

Comment:
Sooo the goal is for this one to be come the king of all eventually right?

Reply:
Exactly